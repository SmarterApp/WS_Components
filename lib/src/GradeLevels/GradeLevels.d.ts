export declare enum GradeLevels {
    NA = 0,
    Grade3 = 1,
    Grade4 = 2,
    Grade5 = 4,
    Grade6 = 8,
    Grade7 = 16,
    Grade8 = 32,
    Grade9 = 64,
    Grade10 = 128,
    Grade11 = 256,
    Grade12 = 512,
    Elementary = 7,
    Middle = 56,
    High = 960,
    All = 1023,
}
export declare class GradeLevel {
    static gradeCaseToString(grade: GradeLevels): string | undefined;
    static gradeCaseToShortString(grade: GradeLevels): string | undefined;
    static gradeLevelToString(grades: GradeLevels): string | undefined;
    static gradeLevelContains(haystack: GradeLevels, needle: GradeLevels): boolean;
    static stringToGradeLevel(gradeString: string): GradeLevels;
    static gradeLevelAdd(grade: GradeLevels, gradeString: string): number;
}
