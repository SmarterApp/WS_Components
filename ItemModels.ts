import * as ItemCard from "./ItemCard";
import * as $ from 'jquery';
import * as GradeLevels from './GradeLevels';
import * as API from './ApiModels';

export interface ScoreSearchParams {
    gradeLevels: GradeLevels.GradeLevels;
    subjects: string[];
    techType: string[];
}

export const ScoreSearchClient = (params: ScoreSearchParams) => API.get<ItemCard.ItemCardViewModel[]>("/ScoringGuide/Search", params) 
