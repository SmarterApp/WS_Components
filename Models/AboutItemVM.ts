import * as Rubric from '../PageTabs/Rubric'
import * as ItemCardViewModel from './ItemCardViewModel'
import * as API from '../Models/ApiModels'

export const ScoreSearchClient = (params: { bankKey: number, itemKey: number }) => API.get<AboutThisItem>("/api/aboutItem", params)

export interface AboutThisItem {
    rubrics: Rubric.Rubric[];
    itemCardViewModel: ItemCardViewModel.ItemCardViewModel;
    depthOfKnowledge: string;
    targetDescription: string;
    commonCoreStandardsDescription: string;
    educationalDifficulty: string;
    evidenceStatement: string;
}