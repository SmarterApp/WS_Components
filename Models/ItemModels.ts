import * as $ from 'jquery';
import { GradeLevels } from './GradeLevels';
import * as API from './ApiModels';
import * as ItemCardViewModel from './ItemCardViewModel';

export interface Subject {
    code: string;
    label: string;
}

export interface TechType extends Subject { }

export interface FilterOptions {
    subjects: Subject[];
    grades: GradeLevels[];
    techTypes: TechType[];
}

export interface ItemFilter extends FilterOptions { };