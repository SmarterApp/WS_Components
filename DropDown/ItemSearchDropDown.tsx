import * as React from 'react';
import * as GradeLevels from "../Models/GradeLevels";
import * as ItemModels from '../Models/ItemModels';

export interface SearchAPIParams {
    itemId: string;
    gradeLevels: GradeLevels.GradeLevels;
    subjects: string[];
    claims: string[];
    interactionTypes: string[];
    performanceOnly: boolean;
}

export interface Props {
    filterOptions: ItemModels.FilterOptions;
    onChange: (params: ItemModels.ItemFilter) => void;
    isLoading: boolean;
    itemFilter: ItemModels.ItemFilter
}

export interface State extends ItemModels.ItemFilter { }

export class ItemSearchDropdown extends React.Component<Props, State>{

    constructor(props: Props) {
        super(props);

        this.state = props.itemFilter;

        this.onChange();
    }
    
    onChange() {
        this.props.onChange(this.state);
    }

    resetFilters() {
        this.setState({ 
            grade: undefined,
            subject: undefined,
            techType: undefined
        }, () => this.onChange());
    }

    keyPressResetFilters(e: React.KeyboardEvent<HTMLElement>) {
        if (e.keyCode === 0 || e.keyCode === 13 || e.keyCode === 32) {
            this.resetFilters();
        }
    }

    changeGrade = (event: React.FormEvent<HTMLSelectElement>) => {
        this.setState({
            grade: Number(event.currentTarget.value)
        }, () => this.onChange());
    }

    renderGrades() {
        let tags = [
            <option key={1} value="All">All</option>
        ];
        this.props.filterOptions.grades.forEach((g, i) => tags.push(
            <option key={i + 2} value={g}>{GradeLevels.caseToString(g)}</option>
        ));

        return (<select value={this.state.grade || "All"} onChange={this.changeGrade}>{tags}</select>);
    }

    changeSubject = (event: React.FormEvent<HTMLSelectElement>) => {
        const selectedCode = event.currentTarget.value;
        let selectedSubject = this.props.filterOptions.subjects.find(s => s.code == selectedCode);

        if (selectedSubject) {
            selectedSubject = {...selectedSubject}
        }
        
        this.setState({
            subject: selectedSubject
        }, () => this.onChange());
        
        const subjectCodes = this.state.subject || [];
    }

    renderSubjects() {
        let tags = [
            <option key={1} value="All">All</option>
        ];
        this.props.filterOptions.subjects.forEach((s, i) => tags.push(
            <option key={i + 2} value={s.code}>{s.label}</option>
        ));

        const val = this.state.subject ? this.state.subject.code : "All";
        return (<select value={val} onChange={this.changeSubject}>{tags}</select>);
    }

    changeTechType = (event: React.FormEvent<HTMLSelectElement>) => {
        const selectedCode = event.currentTarget.value;
        let selectedType = this.props.filterOptions.techTypes.find(t => t.code == selectedCode);

        if (selectedType) {
            selectedType = {...selectedType}
        }

        this.setState({
            techType: selectedType
        }, () => this.onChange());
    }

    renderTechTypes() {
        let tags = [
            <option key={1} value="All">All</option>
        ];
        this.props.filterOptions.techTypes.forEach((t, i) => tags.push(
            <option key={i + 2} value={t.code}>{t.label}</option>
        ));

        const val = this.state.techType ? this.state.techType.code : "All";
        return (<select value={val} onChange={this.changeTechType}>{tags}</select>);
    }

    render() {
        return (
            <div className="search-params">
                <div className="search-header">
                    <div className="search-status">
                        {this.props.isLoading ? <img src="images/spin.gif" className="spin" /> : undefined}
                    </div>

                </div>
                <div className="search-categories" aria-live="polite" aria-relevant="additions removals">
                    {this.renderGrades()}
                    {this.renderSubjects()}
                    {this.renderTechTypes()}
                    <div>
                        <button onClick={() => this.resetFilters()} 
                            onKeyPress={e => this.keyPressResetFilters(e)} 
                            tabIndex={0}>Reset filters</button>
                    </div>
                </div>
            </div>
        );
    }
}
