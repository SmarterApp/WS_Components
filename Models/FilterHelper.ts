import { ItemCardViewModel } from "../Models/ItemCardViewModel";
import { ItemFilter, FilterOptions } from "../Models/ItemModels";
import * as GradeLevels from '../Models/GradeLevels';
import { parseQueryString } from "../Models/ApiModels";
import { AdvancedFilterCategory, AdvancedFilters, OptionType, AdvancedFilterOption } from "../filter/AdvancedFilterModel";

export class FilterHelper {

    //TODO: Get this from the server
    static getFilterOptions(): AdvancedFilters  {
        const subjectsFilterOptions: AdvancedFilterOption[] = [{
            label: "Mathematics",
            key: "MATH",
            isSelected: false,
            type: OptionType.button
        },
        {
            label: "English",
            key: "ELA",
            isSelected: false,
            type: OptionType.button
        }];

        const gradesFilterOptions: AdvancedFilterOption[] = [{
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Elementary),
            key: String(GradeLevels.GradeLevels.Elementary),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade3),
            key: String(GradeLevels.GradeLevels.Grade3),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade4),
            key: String(GradeLevels.GradeLevels.Grade4),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade5),
            key: String(GradeLevels.GradeLevels.Grade5),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade6),
            key: String(GradeLevels.GradeLevels.Grade6),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade7),
            key: String(GradeLevels.GradeLevels.Grade7),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade8),
            key: String(GradeLevels.GradeLevels.Grade8),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.High),
            key: String(GradeLevels.GradeLevels.High),
            isSelected: false,
            type: OptionType.button
        }];

        const techTypesFilterOptions: AdvancedFilterOption[] = [{
            label: "CAT",
            key: "CAT",
            isSelected: false,
            type: OptionType.button
        }, {
            label: "Performance Items",
            key: "PT",
            isSelected: false,
            type: OptionType.button
        }];

        const subjects: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "Subjects",
            helpText: "Subjects HelpText here.",
            filterOptions: [...subjectsFilterOptions],
            displayAllButton: true
        }

        const grades: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "Grade",
            helpText: "Grade HelpText here.",
            filterOptions: [...gradesFilterOptions],
            displayAllButton: true
        }

        const techTypes: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "TechType",
            helpText: "TechType HelpText here.",
            filterOptions: [...techTypesFilterOptions],
            displayAllButton: false
        }

        const value: AdvancedFilters = {subjects, grades, techTypes};
        
        return value;
    }

    static filter(itemCards: ItemCardViewModel[], filter: AdvancedFilterCategory[]): ItemCardViewModel[] {
        const grades = filter.find(afc => afc.label === "Grade");
        const subjects = filter.find(afc => afc.label === "Subjects");
        const techTypes = filter.find(afc => afc.label === "TechType");

        if (grades && grades.filterOptions) {
            let selectedGrades = GradeLevels.GradeLevels.NA;

            grades.filterOptions.forEach(gradeFilter => {
                if (gradeFilter.isSelected) {
                    selectedGrades = selectedGrades | Number(gradeFilter.key);
                }
            });

            if (selectedGrades === GradeLevels.GradeLevels.NA) {
                selectedGrades = GradeLevels.GradeLevels.All;
            }

            itemCards = itemCards.filter(g => GradeLevels.contains(selectedGrades, g.grade));
        }

        if (subjects && subjects.filterOptions) {
            let subjectCode:string|undefined;

            subjects.filterOptions.forEach(s => {
                if(s.isSelected){
                    subjectCode = s.key;
                }
            });
            if(subjectCode !== undefined){
                itemCards = itemCards.filter(i => subjectCode === i.subjectCode);
            }
        }
        //TODO: What is CAT technology? Filter? Ignore?
        if (techTypes && techTypes.filterOptions) {
            if (techTypes.filterOptions.find(t => t.key === "PT" && t.isSelected)) {
                itemCards = itemCards.filter(i => i.isPerformanceItem);
            } else if (techTypes.filterOptions.find(t => t.key === "CAT" && t.isSelected)) {
                itemCards = itemCards.filter(i => !i.isPerformanceItem);
            }
        }

        return itemCards;
    }

    static updateUrl(filter: AdvancedFilterCategory[]) {
        let pairs: string[] = [];

        const grades = filter.find(afc => afc.label === "Grade");
        const subjects = filter.find(afc => afc.label === "Subjects");
        const techTypes = filter.find(afc => afc.label === "TechType");

        if (grades && !grades.disabled) {
            const gradeString: string[] = [];

            grades.filterOptions.forEach(g => {
                if (g.isSelected) {
                    gradeString.push(g.key.toString());
                }
            });

            if (gradeString.length !== 0) {
                gradeString.join(',');
                pairs.push("gradeLevels=" + gradeString);
            }
        }
        if (subjects && !subjects.disabled) {
            const subjString: string[] = [];

            subjects.filterOptions.forEach(s => {
                if (s.isSelected) {
                    subjString.push(s.key);
                }
            });

            if (subjString.length !== 0) {
                subjString.join(',');
                pairs.push("subjects=" + subjString);
            }
        }
        if (techTypes && !techTypes.disabled) {
            const techTypesString: string[] = [];

            techTypes.filterOptions.forEach(t => {
                if (t.isSelected) {
                    techTypesString.push(t.key);
                }
            });
            if (techTypesString.length !== 0) {
                techTypesString.join(',');
                pairs.push("techTypes=" + techTypesString);
            }
        }

        let query: string;
        if (pairs.length === 0) {
            query = "/";
        } else {
            query = "?" + pairs.join("&");
        }
        history.replaceState(null, "", query);
    }

    static readUrl(filterOptions: AdvancedFilters) {
        const queryObject = parseQueryString(window.location.href);
        const subjects = queryObject["subjects"]
            ? queryObject["subjects"]!.map(
                subjCode => {
                    let subject = filterOptions.subjects.filterOptions.find(s => s.key === subjCode);
                    if (subject) {
                        return {
                            code: subject.key,
                            label: subject.label
                        }
                    }
                }) : [];

        const grades = queryObject["grades"]
            ? queryObject["grades"]!
                .map(gradeCode => Number(gradeCode) as GradeLevels.GradeLevels)
            : [];

        const techTypes = queryObject["techTypes"]
            ? queryObject["techTypes"]!.map(typeCode => {
                let techType = filterOptions.techTypes.filterOptions.find(t => t.key === typeCode);
                if (techType) {
                    return {
                        code: techType.key,
                        label: techType.label
                    }
                }
            }) : [];

        return {
            subjects: subjects,
            grades: grades,
            techTypes: techTypes
        } as ItemFilter;
    }
}