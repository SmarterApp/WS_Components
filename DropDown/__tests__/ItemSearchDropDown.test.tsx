import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { ItemSearchDropdown } from '../ItemSearchDropDown';
import * as ItemModels from '../../Models/ItemModels';
import { GradeLevels } from '../../Models/GradeLevels';

describe("ItemFileDropDown", () => {
    const subject: ItemModels.Subject = {
        code: "12547456245",
        label: "234509383405"
    }
    const techType: ItemModels.TechType = subject;
    const subjects: Array<ItemModels.Subject> = [subject]
    const techTypes: Array<ItemModels.Subject> = [techType]

    const filterOptions: ItemModels.FilterOptions = {
        subjects,
        grades: [GradeLevels.All],
        techTypes
    }

    const itemFilter: ItemModels.ItemFilter = {
        subject: subject,
        grade: GradeLevels.All,
        techType: techType
    }

    const props = {
        filterOptions,
        onChange: jest.fn((params: ItemModels.ItemFilter) => {return null}),
        isLoading: false;
        itemFilter,
    }

    it("matches snapshot", () => {
        expect(shallow(<ItemSearchDropdown {...props} />)).toMatchSnapshot()
    })
})