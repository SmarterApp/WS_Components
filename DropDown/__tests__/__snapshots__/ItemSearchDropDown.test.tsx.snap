// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemFileDropDown matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="search-params"
>
    <div
        className="search-header"
    >
        <div
            className="search-status"
        />
        <div>
            <a
                onClick={[Function]}
                onKeyPress={[Function]}
                tabIndex={0}
            >
                Reset filters
            </a>
        </div>
    </div>
    <div
        aria-live="polite"
        aria-relevant="additions removals"
        className="search-categories"
    >
        <select
            onChange={[Function]}
            value={1023}
        >
            <option
                value="All"
            >
                All
            </option>
            <option
                value={1023}
            />
        </select>
        <select
            onChange={[Function]}
            value="12547456245"
        >
            <option
                value="All"
            >
                All
            </option>
            <option
                value="12547456245"
            >
                234509383405
            </option>
        </select>
        <select
            onChange={[Function]}
            value="12547456245"
        >
            <option
                value="All"
            >
                All
            </option>
            <option
                value="12547456245"
            >
                234509383405
            </option>
        </select>
    </div>
</div>,
  "nodes": Array [
    <div
      className="search-params"
>
      <div
            className="search-header"
      >
            <div
                  className="search-status"
            />
            <div>
                  <a
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        tabIndex={0}
                  >
                        Reset filters
                  </a>
            </div>
      </div>
      <div
            aria-live="polite"
            aria-relevant="additions removals"
            className="search-categories"
      >
            <select
                  onChange={[Function]}
                  value={1023}
            >
                  <option
                        value="All"
                  >
                        All
                  </option>
                  <option
                        value={1023}
                  />
            </select>
            <select
                  onChange={[Function]}
                  value="12547456245"
            >
                  <option
                        value="All"
                  >
                        All
                  </option>
                  <option
                        value="12547456245"
                  >
                        234509383405
                  </option>
            </select>
            <select
                  onChange={[Function]}
                  value="12547456245"
            >
                  <option
                        value="All"
                  >
                        All
                  </option>
                  <option
                        value="12547456245"
                  >
                        234509383405
                  </option>
            </select>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ItemSearchDropdown
        filterOptions={
                Object {
                        "grades": Array [
                          1023,
                        ],
                        "subjects": Array [
                          Object {
                            "code": "12547456245",
                            "label": "234509383405",
                          },
                        ],
                        "techTypes": Array [
                          Object {
                            "code": "12547456245",
                            "label": "234509383405",
                          },
                        ],
                      }
        }
        isLoading={false}
        itemFilter={
                Object {
                        "grade": 1023,
                        "subject": Object {
                          "code": "12547456245",
                          "label": "234509383405",
                        },
                        "techType": Object {
                          "code": "12547456245",
                          "label": "234509383405",
                        },
                      }
        }
        onChange={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ItemSearchDropdown {
        "_reactInternalInstance": [Circular],
        "changeGrade": [Function],
        "changeSubject": [Function],
        "changeTechType": [Function],
        "context": Object {},
        "props": Object {
          "filterOptions": Object {
            "grades": Array [
              1023,
            ],
            "subjects": Array [
              Object {
                "code": "12547456245",
                "label": "234509383405",
              },
            ],
            "techTypes": Array [
              Object {
                "code": "12547456245",
                "label": "234509383405",
              },
            ],
          },
          "isLoading": false,
          "itemFilter": Object {
            "grade": 1023,
            "subject": Object {
              "code": "12547456245",
              "label": "234509383405",
            },
            "techType": Object {
              "code": "12547456245",
              "label": "234509383405",
            },
          },
          "onChange": [Function],
        },
        "refs": Object {},
        "state": Object {
          "grade": 1023,
          "subject": Object {
            "code": "12547456245",
            "label": "234509383405",
          },
          "techType": Object {
            "code": "12547456245",
            "label": "234509383405",
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="search-params"
>
          <div
                    className="search-header"
          >
                    <div
                              className="search-status"
                    />
                    <div>
                              <a
                                        onClick={[Function]}
                                        onKeyPress={[Function]}
                                        tabIndex={0}
                              >
                                        Reset filters
                              </a>
                    </div>
          </div>
          <div
                    aria-live="polite"
                    aria-relevant="additions removals"
                    className="search-categories"
          >
                    <select
                              onChange={[Function]}
                              value={1023}
                    >
                              <option
                                        value="All"
                              >
                                        All
                              </option>
                              <option
                                        value={1023}
                              />
                    </select>
                    <select
                              onChange={[Function]}
                              value="12547456245"
                    >
                              <option
                                        value="All"
                              >
                                        All
                              </option>
                              <option
                                        value="12547456245"
                              >
                                        234509383405
                              </option>
                    </select>
                    <select
                              onChange={[Function]}
                              value="12547456245"
                    >
                              <option
                                        value="All"
                              >
                                        All
                              </option>
                              <option
                                        value="12547456245"
                              >
                                        234509383405
                              </option>
                    </select>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="search-params"
>
          <div
                    className="search-header"
          >
                    <div
                              className="search-status"
                    />
                    <div>
                              <a
                                        onClick={[Function]}
                                        onKeyPress={[Function]}
                                        tabIndex={0}
                              >
                                        Reset filters
                              </a>
                    </div>
          </div>
          <div
                    aria-live="polite"
                    aria-relevant="additions removals"
                    className="search-categories"
          >
                    <select
                              onChange={[Function]}
                              value={1023}
                    >
                              <option
                                        value="All"
                              >
                                        All
                              </option>
                              <option
                                        value={1023}
                              />
                    </select>
                    <select
                              onChange={[Function]}
                              value="12547456245"
                    >
                              <option
                                        value="All"
                              >
                                        All
                              </option>
                              <option
                                        value="12547456245"
                              >
                                        234509383405
                              </option>
                    </select>
                    <select
                              onChange={[Function]}
                              value="12547456245"
                    >
                              <option
                                        value="All"
                              >
                                        All
                              </option>
                              <option
                                        value="12547456245"
                              >
                                        234509383405
                              </option>
                    </select>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ItemSearchDropdown
    filterOptions={
        Object {
            "grades": Array [
              1023,
            ],
            "subjects": Array [
              Object {
                "code": "12547456245",
                "label": "234509383405",
              },
            ],
            "techTypes": Array [
              Object {
                "code": "12547456245",
                "label": "234509383405",
              },
            ],
          }
    }
    isLoading={false}
    itemFilter={
        Object {
            "grade": 1023,
            "subject": Object {
              "code": "12547456245",
              "label": "234509383405",
            },
            "techType": Object {
              "code": "12547456245",
              "label": "234509383405",
            },
          }
    }
    onChange={[Function]}
/>,
}
`;
